### PR TITLE
feat(Cypress): Add routing test

### PR DESCRIPTION
**Pull request by Sakilmostak**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Add service level test for Routing

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Tested through Cypress
Screenshot 2024-05-27 at 1 04 52 AM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: cypress-tests/cypress/e2e/RoutingTest/00002-ConnectorMultipleCreate.cy.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  });
+
+  it("stripe-connector-create-call-test", () => {
+    cy.createConnectorWithNameCallTest(
+      createConnectorBody,
+      "stripe",
+      "payment_processor",
+      globalState,
+    );
+  });
```
Comment (SamraatBansal) [inline cypress-tests/cypress/e2e/RoutingTest/00002-ConnectorMultipleCreate.cy.js:34]: Is this function used at multiple places?
If only added for routing test cases still it should be configurable, once discuss this with @vspecky

**Review by SamraatBansal — CHANGES_REQUESTED** [review]

**Review by vspecky — APPROVED** [review]
LGTM

**Review by SamraatBansal — APPROVED** [review]

File: cypress-tests/cypress/e2e/RoutingTest/00000-AccountCreate.cy.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import apiKeyCreateBody from "../../fixtures/create-api-key-body.json";
+import merchantCreateBody from "../../fixtures/merchant-create-body.json";
+import State from "../../utils/State";
+
+let globalState;
+describe("Account Create flow test", () => {
+  before("seed global state", () => {
+    cy.task("getGlobalState").then((state) => {
+      globalState = new State(state);
+      console.log("seeding globalState -> " + JSON.stringify(globalState));
```
Comment (pixincreate) [inline cypress-tests/cypress/e2e/RoutingTest/00000-AccountCreate.cy.js:10]: Can you please remove all the console logs that exposes `globalState`.  
That includes `cy.task("cli_log")` stuffs as well. Let's keep that them only for debugging.

File: cypress-tests/cypress/e2e/RoutingTest/00000-AccountCreate.cy.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    });
+  });
+  after("flush global state", () => {
+    console.log("flushing globalState -> " + JSON.stringify(globalState));
```
Comment (pixincreate) [inline cypress-tests/cypress/e2e/RoutingTest/00000-AccountCreate.cy.js:14]: Here as well.

File: cypress-tests/cypress/e2e/RoutingTest/00001-CustomerCreate.cy.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import customerCreateBody from "../../fixtures/create-customer-body.json";
+import State from "../../utils/State";
+
+let globalState;
+
+describe("Customer Create flow test", () => {
+  before("seed global state", () => {
+    cy.task("getGlobalState").then((state) => {
+      globalState = new State(state);
+      console.log("seeding globalState -> " + JSON.stringify(globalState));
```
Comment (pixincreate) [inline cypress-tests/cypress/e2e/RoutingTest/00001-CustomerCreate.cy.js:10]: Here as well.

File: cypress-tests/cypress/e2e/RoutingTest/00001-CustomerCreate.cy.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    });
+  });
+  after("flush global state", () => {
+    console.log("flushing globalState -> " + JSON.stringify(globalState));
```
Comment (pixincreate) [inline cypress-tests/cypress/e2e/RoutingTest/00001-CustomerCreate.cy.js:14]: Here as well.

File: cypress-tests/cypress/e2e/RoutingTest/00002-ConnectorMultipleCreate.cy.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import createConnectorBody from "../../fixtures/create-connector-body.json";
+import State from "../../utils/State";
+
+let globalState;
+describe("Connector Account Create flow test", () => {
+  before("seed global state", () => {
+    cy.task("getGlobalState").then((state) => {
+      globalState = new State(state);
+      console.log("seeding globalState -> " + JSON.stringify(globalState));
```
Comment (pixincreate) [inline cypress-tests/cypress/e2e/RoutingTest/00002-ConnectorMultipleCreate.cy.js:9]: Here as well.

File: cypress-tests/cypress/e2e/RoutingTest/00002-ConnectorMultipleCreate.cy.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    });
+  });
+
+  after("flush global state", () => {
+    console.log("flushing globalState -> " + JSON.stringify(globalState));
```
Comment (pixincreate) [inline cypress-tests/cypress/e2e/RoutingTest/00002-ConnectorMultipleCreate.cy.js:14]: Here as well.

File: cypress-tests/cypress/e2e/RoutingTest/00003-PaymentRoutingTest.cy.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+  });
+
+  before("seed global state", () => {
+    cy.task("getGlobalState").then((state) => {
+      globalState = new State(state);
+      console.log("seeding globalState -> " + JSON.stringify(globalState));
+      cy.task(
+        "cli_log",
+        "SEEDING GLOBAL STATE -> " + JSON.stringify(globalState),
+      );
```
Comment (pixincreate) [inline cypress-tests/cypress/e2e/RoutingTest/00003-PaymentRoutingTest.cy.js:24]: Remove these

File: cypress-tests/cypress/e2e/RoutingTest/00003-PaymentRoutingTest.cy.js [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+    });
+  });
+
+  afterEach("flush global state", () => {
+    console.log("flushing globalState -> " + JSON.stringify(globalState));
+    cy.task("setGlobalState", globalState.data);
+    cy.task(
+      "cli_log",
+      " FLUSHING GLOBAL STATE -> " + JSON.stringify(globalState),
+    );
```
Comment (pixincreate) [inline cypress-tests/cypress/e2e/RoutingTest/00003-PaymentRoutingTest.cy.js:34]: Remvoe L29 and these.

**Review by pixincreate — CHANGES_REQUESTED** [review]
Other than that, LGTM

**Pull request merged**